### ISSUE TITLE
Filter (all tests) by failure message

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt.fyi#462]
In https://adam.orphis.net/~orphis/wpt.html @orphis built in the ability to filter tests by failure messages, as "Output filter". wpt.fyi should be able to do this, for example searching for "message:TypeError" (strawman!) at the root level should filter to tests where that shows up in the output, such as:
https://wpt.fyi/results/webrtc/RTCRtpParameters-codecs.html?sha=6aef6676d2&label=stable (in Chrome + Edge)

This issue is very similar to https://github.com/web-platform-tests/wpt.fyi/issues/81, but that is about making search work on the results page for a specific tests, while this is about searching for tests that fail in a certain way.

**Comment by lukebjerring** [comment]
@mdittmer - can you confirm whether the `searchcache` includes an index for failure messages?

**Comment by foolip** [comment]
I can't say for certain, but in multiple discussions with @mdittmer I recall the answer being no.

**Comment by Hexcles** [comment]
No. And it'd either increase our memory requirement by several magnitudes or decrease the number of runs in the cache by several magnitudes.

**Comment by foolip** [comment]
FWIW, failure messages were included in my experiment fitting all runs into a few GB of memory.

**Comment by foolip** [comment]
This would have come in handy in https://github.com/web-platform-tests/wpt/pull/20036#issuecomment-550237192.

**Comment by foolip** [comment]
Actually, it's probably about once a week that I find I'd want this but because I know it doesn't work I just move on. Another case today was https://github.com/web-platform-tests/wpt/pull/20109, where I submitted a PR mostly to understand which tests had the subtests in question. I could have just searched by subtest name if it was possible.

@stephenmcgruer FYI in case you also find a recurring need for this.
